Zend/Optimizer/zend_optimizer.c

```diff
@@ -914,6 +914,28 @@ zend_function *zend_optimizer_get_called_func(
 					}
 					return fbc;
 				}
+			} else if (opline->op1_type == IS_CONST
+					   && Z_TYPE_P(CRT_CONSTANT(opline->op1)) == IS_OBJECT
```

Comment:
Objects cannot be `IS_CONST`. I don't think this branch is hit. You should look at the inferred SSA variable instead.